keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -96,28 +95,33 @@ func (co *Common) SetOSInterface(intf types.OSTypeInstaller) {
 	co.OSTypeInstaller = intf
 }
 
-//GetOSVersion gets the OS name
-func GetOSVersion() string {
-	cmd := NewCommand("source /etc/os-release && echo $ID")
+//GetPackageManager get package manager of OS
+func GetPackageManager() string {
+	cmd := NewCommand("command -v apt || command -v yum")
```

Comment:
if `apt` is a alias, for example `alias apt="apt -y"`, `command -v apt` may not end with `apt`. 

```shell
$ docker run --rm -it ubuntu
root@91d3467eecaf:/# command -v apt
/usr/bin/apt
root@91d3467eecaf:/# alias apt="apt -y"
root@91d3467eecaf:/# command -v apt
alias apt='apt -y'
root@91d3467eecaf:/# which apt
/usr/bin/apt
root@91d3467eecaf:/#
```
maybe we could try with `which`

Reply:
We can find the package manager under `/usr/bin/`, most package manager is under this path. And add a parameter in keadm, like `--package-manager` or other. User can provide their package manager to keadm manual.

If keadm have this option, we can provide a interface to user, for example `--package-manage=MYSCRIPT`. The script implement some func like `install mqtt`, . we just use `MYSCRIPT installl mqtt` to install mqtt and other things. user need to complete the install logic, install online or offline, all is ok. :)

Reply:
Refer to https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script, `which` is not a good choice.

An alias does not make whole install cmd like `apt install -y mqtt` failed, besides, we use `sudo apt`, in most production environments, there is little alias in root account. If exists, fail output would raise error to notice users.

We can auto detect package manager cmd, no need to ask users to specify it. Also, install by  package manager still need network in most cases, it is outside this pr indeed.